passes/pmgen/ice40_dsp.pmg

```diff
@@ -1,3 +1,5 @@
+pattern ice40_dsp
```

Comment:
What does this new `pattern` keyword do?
(Does it need to go into the doc?)

Reply:
Yes, added to the doc now. I'm also planning on adding `subpattern` statements as well as `namespace` and `endnamespace`. I'll update the doc accordingly as soon as I add them.